### ISSUE TITLE
Use only one of the optimizers as warm-up 

### ISSUE THREAD

**Issue by Brechard** [PyTorchLightning/pytorch-lightning#1902]
#### What is your question?
Hi! I am trying to train a GAN and I would like to train only the Discriminator for the first x steps, how can that be done? 

I've seen the code for step optimizers at arbitrary intervals but it is not clear to me how it works (it says update generator every 2 steps and discriminator every 4, but then do they cumulate the gradients until they update or what happens in the odd steps? Why are they cumulating gradients then?) so I ended up using the frequency in the dictionary returned in the configure_optimizers functions which gives the behaviour I wanted of train x times optimizer A, train Y times optimizer B. Now I am looking to do for the first S steps only train optimizer A, then train x times A, train Y times B.

#### What's your environment?

 - OS: Win
 - Packaging: conda
 - Version: 0.7.6rc2

**Comment by MaharshiYeluri01** [comment]
In the training_step function you have access to the bath iteration number(batch_nb), you can ignore the back prop step for generator for first n iterations
Usually batch size while training the GAN's is less(because of size and multiple backward functions) so to eliminate the noisy updates, gradients are accumulated for some mini batches before the optimizer steps

**Comment by Brechard** [comment]
thanks for the reply, how do I ignore the backprop step though?

**Comment by sebastienwood** [comment]
You can override the optimizer step like so
```
def optimizer_step(self, current_epoch, batch_nb, optimizer, optimizer_i, second_order_closure=None):
        if optimizer_i == 0:
            optimizer.step()
            optimizer.zero_grad()
```
Notice you can have any kind of condition to trigger the optimizer step.
